### PR TITLE
added button to add people, added diff enviorments, refactored conten…

### PR DESCRIPTION
**Pull request by littlejohntj**
…tview

### CODE REVIEW

File: ComposableFirebase/ContentView.swift [hunk 1]
```diff
@@ -25,12 +25,15 @@ struct PersonClient {
...
 struct AppState: Equatable {
-  var people = [Person]()
+    var people = [Person]()
+    var isShowingSheet = false
 }
 
 enum AppAction: Equatable {
-  case beginListening
-  case peopleUpdated(Result<PersonClient.Action, PersonClient.Error>)
+    case startAddPersonFlow
```
Comment (technoplato) [inline ComposableFirebase/ContentView.swift:33]: 👍 

Love leaving notes to reference back in PRs

We just discussed this and you had the idea to separate the abstract concept of `startAddPersonFlow` away from `setShowingSheet`. 

This is great and in the spirit of composability, as it allows us to decide later (deferring implementation details) exactly what `startAddPersonFlow` looks like

File: ComposableFirebase/ContentView.swift [hunk 2]
```diff
@@ -42,44 +45,64 @@ let peopleReducer = Reducer<AppState, AppAction, AppEnvironment> {state, action,
...
+            }
+            .navigationTitle("Dummy")
+            .toolbar {
+                Button {
+                    viewStore.send(.startAddPersonFlow)
+                } label: {
+                    Image(systemName: "plus")
+                }
+            }
+            .sheet(isPresented: viewStore.binding(get: { appState in
```
Comment (technoplato) [inline ComposableFirebase/ContentView.swift:86]: Nice job getting this right from memory

Here's the ultra clean way from their repo: https://github.com/pointfreeco/swift-composable-architecture/blob/main/Examples/CaseStudies/SwiftUICaseStudies/03-Navigation-Sheet-LoadThenPresent.swift#L82-L84

Comment (technoplato) [reply ComposableFirebase/ContentView.swift:86]: Screen Shot 2021-08-22 at 12 31 19 PM

File: ComposableFirebase/ContentView.swift [hunk 3]
```diff
@@ -42,44 +45,64 @@ let peopleReducer = Reducer<AppState, AppAction, AppEnvironment> {state, action,
...
   }
 }
 
-  //struct ContentView_Previews: PreviewProvider {
-  //    static var previews: some View {
-  //        ContentView()
-  //    }
-  //}
-  //
+//  struct ContentView_Previews: PreviewProvider {
```
Comment (technoplato) [inline ComposableFirebase/ContentView.swift:98]: Worth exploring how to get our previews working

File: ComposableFirebase/Enviorment.swift [hunk 4]
```diff
@@ -0,0 +1,16 @@
+//
+//  Enviorments.swift
+//  Enviorments
+//
+//  Created by TJ Littlejohn on 8/22/21.
+//
+
+import Foundation
+
+struct Environment {
```
Comment (technoplato) [inline ComposableFirebase/Enviorment.swift:10]: I'm going to tinker for 5 minutes to see if we make this an Extension on AppEnvironment if we can use that dot syntax. I really like that. It's one of my favorite things about Swift.

**Review by technoplato — APPROVED** [review]

**Comment by technoplato** [comment]
Merge https://github.com/technoplato/FirebaseComposable/pull/2 after this is merged

File: ComposableFirebase/ContentView.swift (hunk 3 above)
Comment (littlejohntj) [reply ComposableFirebase/ContentView.swift:98]: yeah I'm not sure why it didn't :/ I wonder if just having firebase in the project fucks that up? This code should have nothing to do with firebase

File: ComposableFirebase/ComposableFirebaseApp.swift [hunk 5]
```diff
@@ -22,10 +22,7 @@ struct ComposableFirebaseApp: App {
               store: Store(
                 initialState: .init(),
                 reducer: peopleReducer,
-                environment: AppEnvironment(
-                  personClient: .live,
-                  mainQueue: .main
-                )
+                environment: Environment.offline
```
[the remark this answers is not in the document]
Comment (littlejohntj) [reply ComposableFirebase/ComposableFirebaseApp.swift]: :)

File: ComposableFirebase/ContentView.swift [hunk 6]
```diff
@@ -62,24 +64,34 @@ struct ContentView: View {
   
   var body: some View {
     WithViewStore(self.store) { viewStore in
-      VStack {
-        
-        Text("Dummy")
         
-        List( viewStore.people ) { person in
-          Text(person.name)
+        NavigationView {
```
[the remark this answers is not in the document]
Comment (littlejohntj) [reply ComposableFirebase/ContentView.swift:74]: thx

File: ComposableFirebase/ContentView.swift (hunk 1 above)
Comment (littlejohntj) [reply ComposableFirebase/ContentView.swift:33]: this is great, love that

File: ComposableFirebase/ContentView.swift (hunk 2 above)
Comment (littlejohntj) [reply ComposableFirebase/ContentView.swift:86]: damn missed that case path I think right?

Comment (littlejohntj) [reply ComposableFirebase/ContentView.swift:86]: yeah I did a diff one, damn so close! love this though

File: ComposableFirebase/ContentView.swift (hunk 3 above)
Comment (technoplato) [reply ComposableFirebase/ContentView.swift:98]: I actually ran the preview live with Firebase. 

We can just timebox 15 minutes at some point and resolve it

**Pull request merged**